ui/src/components/pages/Namespace/partials/PipelineCard/index.tsx

```diff
@@ -74,6 +90,61 @@ export function PipelineCard({
   const handleDeleteChange = useCallback((event: SelectChangeEvent<string>) => {
     setDeleteOption(event.target.value);
   }, []);
+  const handlePlayClick = useCallback((e) => {
+    setPayloadSet(true);
```

Comment:
Instead of tracking two state items, just have `statusPayload` be undefined by default, then anytime it changes the fetch hook can run using its latest value.